crates/sparrow-batch/src/batch.rs

```diff
@@ -77,6 +78,35 @@ impl Batch {
         })
     }
 
+    /// Updates the up_to_time.
+    pub fn with_up_to_time(self, up_to_time: RowTime) -> Self {
+        Batch {
+            data: self.data,
+            up_to_time,
+        }
+    }
+
+    /// Replaces the batch's data with the given data.
+    ///
+    /// Panics if the [BatchInfo] is empty.
+    pub fn with_data(self, data: ArrayRef) -> Self {
+        if let Some(info) = self.data {
+            Self {
+                data: Some(BatchInfo {
+                    data,
+                    time: info.time,
+                    subsort: info.subsort,
+                    key_hash: info.key_hash,
+                    min_present_time: info.min_present_time,
+                    max_present_time: info.max_present_time,
+                }),
+                up_to_time: self.up_to_time,
```

Comment:
Similarly here -- I wonder if we should check the affect this has on the relationship between times int he data and the up to time?

Reply:
Use `BatchInfo::new` for validations